### PR TITLE
Update the server download journey

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
- Updated the order to Multipass -> Manual -> MAAS
- Move the CTA button below the option buttons
- Update the content across the three pages
- Added anchors to the manual download page

## QA

- Go to the demo and go to /download/server.
- Check the content matches the [copy doc](https://docs.google.com/document/d/12kr3JvN3ALMdr5IPOheMwOaf-ygvEulFo5D7x-0ag0k/edit)
- Click through the steps checking the content
- Ensure the download works
- Ensure all navigation works

## Issue / Card
Fixes https://github.com/canonical-web-and-design/web-squad/issues/3608

### CODE REVIEW

**Comment by SirSamTumless** [comment]
@tytus-kurek Ant when and did the work last night. Please could you review!

**Comment by pmahnke** [comment]
I think the green button should be above the navigation buttons with some space between.

I also think the h2 > h5 > h1 > h4 heads are a bit all over the place visually... not sure what to recommend there.

image

**Comment by anthonydillon** [comment]
@pmahnke where are you seeing that screenshot? I see:
image

**Comment by pmahnke** [comment]
I hacked that together as an alternative?

File: templates/download/server/step2.html [hunk 1]
```diff
@@ -1,53 +1,216 @@
...
+      <p>BitTorrent sometimes enables higher download speeds and more reliable downloads of large files.</p>
+      <ul class="p-list">
+        <li class="p-list__item">
+          <a class="download-torrent" href="https://releases.ubuntu.com/{{ releases.lts.short_version }}/ubuntu-{{ releases.lts.full_version }}-live-server-amd64.iso.torrent">
+          Ubuntu Server {{ releases.lts.full_version }} LTS&nbsp;&rsaquo;
+        </a>
+        </li>
+        <li class="p-list__item">
+          <a class="download-torrent" href="https://releases.ubuntu.com/{{ releases.latest.short_version }}/ubuntu-{{ releases.latest.full_version }}-live-server-amd64.iso.torrent">
+          Ubuntu Server {{ releases.latest.full_version }}&nbsp;&rsaquo;
```
Comment (matthewpaulthomas) [inline templates/download/server/step2.html:152]: These are inconsistent with the eight other “›” links on this page, which go to Web pages; and inconsistent with the following card, where a direct download is styled as a button (with no visible hint that it actually needs to be one). These should be buttons too.

**Review by matthewpaulthomas — DISMISSED** [review]

File: templates/download/server/choose.html [hunk 2]
```diff
@@ -1,203 +1,63 @@
...
-      </p>
-      <form action="/download/server" method="post" style="display: inline">
-        <input type="hidden" name="version" value="{{ releases.latest.full_version }}">
-        <input type="hidden" name="architecture" value="amd64">
-        <input type="hidden" name="next-step" value="download">
-        <button type="submit" class="p-button--positive" onclick="dataLayer.push({'event' : 'GAEvent', 'eventCategory' : 'Server download', 'eventAction' : 'Select option', 'eventLabel' : 'Manual install - latest', 'eventValue' : undefined });">Download Ubuntu Server {{ releases.latest.full_version }}</button>
-      </form>
+      <p class="p-muted-heading">Option 3</p>
+      <h1 class="js-download-option">Automated server provisioning</h1>
+      <p class="p-heading--four">Provision machines with Ubuntu Server in a fully automated way with MAAS</p>
```
[the remark this answers is not in the document]
Comment (anasereijo) [reply templates/download/server/choose.html:18]: @anthonydillon this will make it look much better

**Comment by tytus-kurek** [comment]
Hello,

Great progress so far!

Some minor suggestions before we go live:
- Would it be possible to have the option buttons always at a fixed position on the page? At the moment their position is moving slightly when switching between the options.
- I think we agreed that the CTA button will be moving depending on which option the user is. Option 1 - the CTA button under the Option 1 button; Option 2 - the CTA button under the Option 2 button, etc.
- Would it be possible to have all the option buttons and the CTA button of the same size? Would look better IMHO.
- Can we add some more white spaces between the option buttons?
- Ideally, I'd like the columns aligned in the way that all top headings: "Instant Ubuntu VMs", "Manual server installation" and "Automated server provisioning" would show up in one line.
- On Option 2, can we rename "other" to "alternative", please and add a link to alternative downloads too?

Cheers

File: templates/download/server/choose.html [hunk 3]
```diff
@@ -1,203 +1,63 @@
...
-      </div>
-    </div>
-  </div>
-  {% if releases.latest.short_version > releases.lts.short_version %}
-  <div class="p-strip is-shallow">
-    <div class="u-fixed-width">
-      <hr>
+  <div class="row">
+    <div class="col-12">
+      <h2>Get Ubuntu Server!</h2>
```
[the remark this answers is not in the document]
Comment (anthonydillon) [reply templates/download/server/choose.html]: done.

File: templates/download/server/choose.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (anthonydillon) [reply templates/download/server/choose.html]: Done

File: templates/download/server/step2.html [hunk 4]
```diff
@@ -1,53 +1,216 @@
...
+    <div class="col-7">
+      <p class="p-muted-heading">
+        Option 2
+      </p>
+      <h1 class="js-download-option">
+        Manual server installation
+      </h1>
+      <h2 class="p-heading--three">
+        USB or DVD image based physical install
+      </h2>
```
[the remark this answers is not in the document]
Comment (anthonydillon) [reply templates/download/server/step2.html:15]: Fixed in the heading update commit

File: templates/download/server/choose.html [hunk 5]
```diff
@@ -1,203 +1,63 @@
...
-    <div class="col-4 p-card">
-      <h3 class="p-card__title">Ubuntu for IBM Z</h3>
-      <p>
-        IBM Z and LinuxONE leverage open technology solutions to meet the demands of the new application economy. Ubuntu is now available, with Juju and Ubuntu OpenStack.
-      </p>
-      <p>
-        <a href="/download/server/s390x">Get Ubuntu for IBM Z&nbsp;&rsaquo;</a>
-      </p>
+    <div class="col-12">
+      <p><a href="https://maas.io" class="p-button--positive p-link--external u-no-margin--top" onclick="dataLayer.push({'event' : 'GAEvent', 'eventCategory' : 'Server download', 'eventAction' : 'Select option', 'eventLabel' : 'MAAS', 'eventValue' : undefined });">Get MAAS</a></p>
```
[the remark this answers is not in the document]
Comment (anthonydillon) [reply templates/download/server/choose.html]: I have moved the CTA buttons above the navigation/option buttons

File: templates/download/server/choose.html [hunk 6]
```diff
@@ -1,203 +1,63 @@
...
-      <p><a href="/download/alternative-downloads">See alternative downloads&nbsp;&rsaquo;</a></p>
+      <form action="/download/server" method="post" style="display: inline">
+        <input type="hidden" name="next-step" value="step2">
+        <button type="submit" class="p-button--neutral" onclick="dataLayer.push({'event' : 'GAEvent', 'eventCategory' : 'Server download', 'eventAction' : 'Next option', 'eventLabel' : 'Option 2', 'eventValue' : undefined });">
+          Option 2 - Manual server installation
+        </button>
+      </form>
+      <button disabled class="p-button--neutral u-no-margin--left u-no-margin--top">
+        Option 3 - Automated server provisioning
+      </button>
```
[the remark this answers is not in the document]
Comment (anthonydillon) [reply templates/download/server/choose.html:51]: The original idea was to "force" everyone to visit each before downloading but that is no longer true but a change that would have to be signed off I believe

File: templates/download/server/step2.html [hunk 7]
```diff
@@ -1,53 +1,216 @@
...
       <ul class="p-list">
-        <li class="p-list__item is-ticked">MAAS turns your data centre into a bare metal cloud</li>
-        <li class="p-list__item is-ticked">On-demand physical machines with Ubuntu Server, Windows, CentOS and more</li>
-        <li class="p-list__item is-ticked">Infrastructure discovery and automation</li>
+        <li class="p-list__item is-ticked">OS security guaranteed until April 2025</li>
+        <li class="p-list__item is-ticked">Extended security maintenance until April 2030</li>
+        <li class="p-list__item is-ticked">Commercial support for enterprise customers</li>
       </ul>
+      <p>
+        The default is Ubuntu Server 20.04.1 LTS Live Install on x86.
```
[the remark this answers is not in the document]
Comment (anthonydillon) [reply templates/download/server/step2.html:25]: Good idea, I will move them up.

File: templates/download/server/step2.html (hunk 1 above)
Comment (anthonydillon) [reply templates/download/server/step2.html]: Done

**Review by pmahnke — APPROVED** [review]
+1

File: templates/download/server/step2.html (hunk 1 above)
Comment (pmahnke) [reply templates/download/server/step2.html]: I think this looks much better.

**Comment by anthonydillon** [comment]
@tytus-kurek 
>Would it be possible to have the option buttons always at a fixed position on the page? At the moment their position is moving slightly when switching between the options.

This is not possible as the content above is different and wraps different on different screen sizes. I have tried to align it as much as I can.

> I think we agreed that the CTA button will be moving depending on which option the user is. Option 1 - the CTA button under the Option 1 button; Option 2 - the CTA button under the Option 2 button, etc.

We have moved the CTA button above the options to sit with the content about that CTA as requested by UX.

> Would it be possible to have all the option buttons and the CTA button of the same size? Would look better IMHO.

Not without making all the button a third of the screen

> Can we add some more white spaces between the option buttons?

Done, thanks.

> Ideally, I'd like the columns aligned in the way that all top headings: "Instant Ubuntu VMs", "Manual server installation" and "Automated server provisioning" would show up in one line.

UX requested the Option #: be moved back in front of the title which makes this more difficult.

> On Option 2, can we rename "other" to "alternative", please and add a link to alternative downloads too?

Done, thanks.

**Comment by tytus-kurek** [comment]
@anthonydillon 

> This is not possible as the content above is different and wraps different on different screen sizes. I have tried to align it as much as I can.

Understood.

> We have moved the CTA button above the options to sit with the content about that CTA as requested by UX.

Sounds reasonable, but I can imagine 90% of people will click the green button without even reading the content. I think having it above the option buttons will result in most of the people not even noticing that there are multiple options available.

> Not without making all the button a third of the screen

Understood. Would that really hurt, though? On my screen, it's already 85% of the width.

> UX requested the Option #: be moved back in front of the title which makes this more difficult.

What's the rationale behind this change?

> Done, thanks.

I really like how those "alternative" things show up on the screen now.

**Comment by matthewpaulthomas** [comment]
This is looking much better, thanks.

>> We have moved the CTA button above the options to sit with the content about that CTA as requested by UX.
>
> Sounds reasonable, but I can imagine 90% of people will click the green button without even reading the content. I think having it above the option buttons will result in most of the people not even noticing that there are multiple options available.

I agree that’s a problem. However, it’s a fundamental of UI layout that changing the value/selection in a control may alter stuff below the control, or (much less often) above the control, but not both.

Separately [I suggested changing the buttons to tabs](https://github.com/canonical-web-and-design/ubuntu.com/pull/8978#discussion_r552677249). It wasn’t for this reason, but tabs above the section contents would also be much more prominent than buttons further down (and also quicker to read, because they wouldn’t need “Option N:” prefixes).

>> UX requested the Option #: be moved back in front of the title which makes this more difficult.
>
> What's the rationale behind this change?

As [I explained above](https://github.com/canonical-web-and-design/ubuntu.com/pull/8978#discussion_r552654489), this simplifies the headings at the start of the page contents — from from four out-of-order heading levels, to two correctly-ordered heading levels.

**Comment by SirSamTumless** [comment]
@tytus-kurek and @anasereijo What's the latest on this?

**Comment by tytus-kurek** [comment]
@SirSamTumless LGTM

**Comment by anasereijo** [comment]
@SirSamTumless this is looking much better now.

**Review by jkfran — APPROVED** [review]

**Pull request merged**